src/Mod/Arch/archobjects/reinforcement_linear.py

```diff
@@ -0,0 +1,343 @@
+                "ArrayOfRebars",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "The distance between the start of the "
+                        "reinforcement and the first bar (concrete cover)."
+                    )
+                ),
+            )
+        obj.OffsetStart = 0
+
+        # concrete cover end
+        if "OffsetEnd" not in pl:
+            obj.addProperty(
+                "App::PropertyLength",
+                "OffsetEnd",
+                "ArrayOfRebars",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "The distance between the end of the "
+                        "reinforcement and the last bar (concrete cover)."
+                    )
+                ),
+            )
+        obj.OffsetEnd = 0
+
+        # space between rebars
+        if "Spacing" not in pl:
+            obj.addProperty(
+                "App::PropertyLength",
+                "Spacing",
+                "ArrayOfRebars",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    "The spacing between the bars"
+                )
+            )
+        obj.Spacing = 200
+
+        # fixed value
+        # this value will fixed if one of the others is changed
+        # the third will be calculated thus:
+        # one value will be changed by the user
+        # one value is fixed by this attribute
+        # one value will be recalculated onChanged()
+        if "FixedAttribut" not in pl:
+            obj.addProperty(
+                "App::PropertyEnumeration",
+                "FixedAttribut",
+                "ArrayOfRebars",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "The value which is fixed whereas one of the others "
+                        "is changed and the third will be recalculated."
+                    )
+                )
+            )
+        obj.FixedAttribut = [
+            "Amount",
+            "Distance",
+            "Spacing"
+        ]
+        obj.FixedAttribut = "Distance"
+
+        """
+        # AddRestTo
+        # if ChangeableValue is Amount than there could be some rest
+        # it is to choose what should happen with rest
+        obj.addProperty(
+            "App::PropertyEnumeration",
+            "AddRestTo",
+            "ArrayOfRebars",
+            "Where should the rest be added if ChangeableValue is Amount."
+        )
+        obj.AddRestTo = [
+            "OffsetStart_and_OffsetEnd",
+            "OffsetStart",
+            "OffsetEnd",
+            "Adapt_Distance"
+        ]
+        obj.AddRestTo = "OffsetEnd"
+        """
+
+    def onChanged(
+        self,
+        obj,
+        prop
+    ):
+
+        # since we overwrite the method we need to explicit call it
+        ArchComponent.Component.onChanged(self, obj, prop)
```

Comment:
`super()`

Reply:
the use of super() returns an error.

Reply:
It should use proper syntax, like super(Reinforcement, self).onChanged(obj, prop). Or do you mean this also fails?

Reply:
I will try once again. I like super() too.

Reply:
works :-)